### PR TITLE
CI enhancements/shell script cleanup

### PR DESCRIPTION
**Pull request by cpu**
## Ensure early build exit for failed CI scripts

Per Travis docs the way to have the build exit if one script fails is
to only specify one script and rely on shell operators to short circuit
on non-zero exit. Kind of ugly... but ok Travis............ if you say
so..........................

Resolves https://github.com/jlund/streisand/issues/725

## Refactor/Cleanup test/test.sh

This PR updates `tests/tests.sh` to have less duplication & to pass
a shellcheck invocation. The commonality between it and the base
`streisand` script was extracted to `util/ansible_check.sh` and included
in both places. To make shellcheck follow the includes we add a `-x` to
the `.travis.yml` invocation of shellcheck. A handful of if statements were
replaced by a case statement. Common ansible-playbook invocation
was refactored to a function.

## Document CI

This PR adds some high level documentation about CI to 
`documentation/testing.md` to hopefully make the testing more 
accessible to external contributors.

### CODE REVIEW

**Comment by cpu** [comment]
Seems like I broke something :blush: - I'll take a look shortly.

File: documentation/testing.md [hunk 1]
```diff
@@ -1,3 +1,53 @@
...
+tests](https://travis-ci.org/jlund/streisand). It works by installing required
+test tools (e.g. shellcheck) and an Ansible environment on a Ubuntu Trusty
+(14.04) machine.
+
+The `test.sh` wrapper script
+--------------------------------
+
+The `test/test.sh` wrapper script is invoked to perform tests and supports
+a argument for specifying what actions should be taken. The following are
+supported arguments:
```
Comment (alimakki) [inline documentation/testing.md]: a argument -> an argument

File: documentation/testing.md [hunk 2]
```diff
@@ -1,3 +1,53 @@
...
+
+* **"syntax"** checks the Streisand playbooks for Ansible syntax errors.
+
+* **"run"** runs the CI tests. It assumes the local environment is already
+    prepared from a previous "setup" run.
+
+* **"ci"** combines "setup" and "run".
+
+* **"ful""** performs the same as "ci" but additionally adds verbose output to
+    the Ansible run. This is very verbose but can be useful for diagnosing
```
Comment (alimakki) [inline documentation/testing.md]: ful -> full

**Comment by cpu** [comment]
@alimakki Thanks for the review - typos are all fixed in 0450d33d7cd351907eb633b3cc93a69ebfd731e0

File: documentation/testing.md (hunk 2 above)
Comment (alimakki) [inline documentation/testing.md]: Looks like there's an extra double quote after full, otherwise all good

Comment (cpu) [inline documentation/testing.md]: Oops! good catch, thanks! Fixed in	2a5d436

**Comment by cpu** [comment]
Thanks for the reviews! I'll merge when CI is finished from 2a5d436

**Pull request merged**